src/mediapipe_internal/mediapipegraphconfig.hpp

```diff
@@ -0,0 +1,177 @@
+        const std::string& graphPath = "",
+        const bool passKfsRequest = false) :
+        graphName(graphName),
+        graphPath(graphPath),
+        passKfsRequest(passKfsRequest) {
+    }
+
+    void clear() {
+        graphName.clear();
+        graphPath.clear();
+        passKfsRequest = false;
+    }
+
+    /**
+         * @brief Get the Graph name
+         *
+         * @return const std::string&
+         */
+    const std::string& getGraphName() const {
+        return this->graphName;
+    }
+
+    /**
+         * @brief Set the Graph name
+         *
+         * @param name
+         */
+    void setGraphName(const std::string& graphName) {
+        this->graphName = graphName;
+    }
+
+    /**
+         * @brief Get the Graph Path
+         *
+         * @return const std::string&
+         */
+    const std::string& getGraphPath() const {
+        return this->graphPath;
+    }
+
+    /**
+         * @brief Set the Graph Path
+         *
+         * @param graphPath
+         */
+    void setGraphPath(const std::string& graphPath) {
+        this->graphPath = graphPath;
+
+        if (!FileSystem::isLocalFilesystem(graphPath)) {
+            // Cloud filesystem
+            this->graphPath = graphPath;
+        } else if (graphPath.at(0) == '/') {
+            // Full path case
+            this->graphPath = graphPath;
+        } else {
+            // Relative path case
+            if (this->rootDirectoryPath.empty())
+                throw std::logic_error("Using graph relative path without setting configuration directory path.");
+            this->graphPath = this->rootDirectoryPath + graphPath;
+        }
+    }
+
+    /**
+         * @brief Set root directory path
+         *
+         * @param rootDirectoryPath
+         */
+    void setRootDirectoryPath(const std::string& rootDirectoryPath) {
+        this->rootDirectoryPath = rootDirectoryPath;
+    }
+
+    /**
+         * @brief Get the passKfsRequest
+         *
+         * @return const std::string&
+         */
+    const bool getPassKfsRequestFlag() const {
+        return this->passKfsRequest;
+    }
+
+    /**
+         * @brief Set the Config Path
+         *
+         * @param configPath
+         */
+    void setPassKfsRequestFlag(const bool passKfsRequest) {
+        this->passKfsRequest = passKfsRequest;
+    }
+
+    /**
+     * @brief  Parses all settings from a JSON node
+        *
+        * @return Status
+        */
+    Status parseNode(const rapidjson::Value& v) {
+        try {
+            this->setGraphName(v["name"].GetString());
+            this->setGraphPath(v["graph_path"].GetString());
+            if (v.HasMember("graph_pass_kfs_request"))
+                this->setPassKfsRequestFlag(v["graph_pass_kfs_request"].GetBool());
+            else
+                this->setPassKfsRequestFlag(false);
+        } catch (std::logic_error& e) {
+            SPDLOG_DEBUG("Relative path error: {}", e.what());
```

Comment:
When this error happens?

Reply:
In setGraphPath if we change the implementation and the root path was not set in the relative path scenario.